fw/sock_clnt.c

```diff
@@ -672,13 +679,19 @@ tfw_sock_clnt_start(void)
 		list_del(&ls->list);
 		list_add(&ls->list, &tfw_listen_socks);
 
+		/*
+		 * Paired with tfw_classify_conn_estab(): firstly add the port
+		 * to the bitmap and then move it to the listen state to
+		 * guarantee that the HTTP limits initialization code was called.
+		 */
+		tfw_classifier_add_inport(tfw_addr_port(&ls->addr));
+
 		if ((r = tfw_listen_sock_start(ls))) {
 			T_ERR_ADDR("can't start listening on", &ls->addr,
 				   TFW_WITH_PORT);
 			goto done;
```

Comment:
When error occur at this place opened sockets will hang and they won't be closed. `tfw_sock_clnt_stop()` will not be called, because module has not been started yet. `tfw_cfgop_cleanup_sock_clnt()` doesn't close sockets too, also it doesn't free `TfwListenSock` which stored in `listen_socks_array`.